### PR TITLE
CXX-626: Added conversion between b_date and std::chrono::system_cloc…

### PR DESCRIPTION
**Pull request by oferei**
solved CXX-626. requires review.

### CODE REVIEW

File: src/bsoncxx/types.hpp [hunk 1]
```diff
@@ -201,9 +201,21 @@ BSONCXX_INLINE bool operator==(const b_bool& lhs, const b_bool& rhs) {
 struct BSONCXX_API b_date {
     static constexpr auto type_id = type::k_date;
 
+    BSONCXX_INLINE
+    explicit b_date(int64_t value)
+        : value(value) {}
+
+    BSONCXX_INLINE
+    explicit b_date(const std::chrono::system_clock::time_point& tp)
```
Comment (amidvidy) [inline src/bsoncxx/types.hpp]: I think you need to add a `<chrono>` include.

**Comment by amidvidy** [comment]
I think travis is failing due to the missing include. Could you add it so we can check if your test passes CI?

Otherwise this looks pretty good! Thanks for contributing.

File: src/bsoncxx/test/bson_b_date.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    system_clock::time_point now1, now2;
+
+    now1 = system_clock::now();
+    b_date d{now1};
+    now2 = d;
+
+    //std::cout << "now1: " << time_point_cast<milliseconds>(now1).time_since_epoch().count() << std::endl;
+    //std::cout << "now2: " << time_point_cast<milliseconds>(now2).time_since_epoch().count() << std::endl;
+
+    REQUIRE(time_point_cast<milliseconds>(now1) == time_point_cast<milliseconds>(now2));
```
Comment (amidvidy) [inline src/bsoncxx/test/bson_b_date.cpp]: can we also add a test that creates two b_date objects from the same time_point and then checks that their `value` members are equal?

**Comment by amidvidy** [comment]
Also, I think it is fine to only accept system_clock::time_point. steady_clock and high_resolution_clock are both pretty unsuitable for this purpose - if someone needs that we can add it later.

File: src/bsoncxx/test/bson_b_date.cpp (hunk 2 above)
Comment (oferei) [inline src/bsoncxx/test/bson_b_date.cpp]: yeah, sure.

File: src/bsoncxx/test/bson_b_date.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    using std::chrono::milliseconds;
+
+    system_clock::time_point now1, now2;
+
+    now1 = system_clock::now();
+    b_date d{now1};
+    now2 = d;
+
+    //std::cout << "now1: " << time_point_cast<milliseconds>(now1).time_since_epoch().count() << std::endl;
+    //std::cout << "now2: " << time_point_cast<milliseconds>(now2).time_since_epoch().count() << std::endl;
```
Comment (amidvidy) [inline src/bsoncxx/test/bson_b_date.cpp]: also, please remove these print statements :)

File: src/bsoncxx/types.hpp [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #include <bsoncxx/config/prelude.hpp>
 
 #include <cstring>
+#include <chrono>
```
Comment (amidvidy) [inline src/bsoncxx/types.hpp]: This is a bit of a nitpick, but for each header subsection we prefer to sort them - so <chrono> goes before <cstring>

**Comment by amidvidy** [comment]
Great, now can you just squash this in to one commit with the original `CXX-626 ....` commit message, and then I'll merge it. Thanks again!

File: src/bsoncxx/types.hpp (hunk 1 above)
Comment (oferei) [inline src/bsoncxx/types.hpp]: yup. funny that I could build it locally.

**Comment by oferei** [comment]
um, how do I squash?

**Comment by amidvidy** [comment]
The easiest way is to do an interactive rebase. 

e.g. `git rebase -i origin/master`

Whatever you configured $EDITOR to be in your environment should pop up with a list of commits. Then you leave the 'c' in front of the first commit, and change the other lines to start with 's' so they get "squashed" in to the first commit. More info: https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history

**Comment by amidvidy** [comment]
note that after you squash you will need to force-push to your feature branch.

**Comment by oferei** [comment]
squashed.

**Comment by amidvidy** [comment]
Great! going to merge now.

**Comment by oferei** [comment]
awesome. glad I could help.

**Pull request merged**